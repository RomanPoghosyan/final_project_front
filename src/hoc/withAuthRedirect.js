import React from "react";
import {useSelector} from "react-redux";
import {Redirect} from "react-router-dom";

const withAuthRedirect = (allowUnAuthorized) => (Component) => (props) => {
    const isAuth = useSelector(state => state.user.isAuth);
    if((allowUnAuthorized && !isAuth) || (!allowUnAuthorized && isAuth)) return <Component {...props} />;
    return <Redirect to={"/"} />;

};

export default withAuthRedirect;