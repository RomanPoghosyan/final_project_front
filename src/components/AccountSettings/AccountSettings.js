import React from "react";
import {makeStyles} from "@material-ui/core/styles";
import AccountSettingsForm from "./AccountSettingsForm/AccountSettingsForm";
import {useDispatch, useSelector} from "react-redux";
import {updateUser} from "../../redux/User/actions";
import withAuthRedirect from "../../hoc/withAuthRedirect";

const useStyles = makeStyles(theme => ({
    wrapper: {
      width: "100%",
      height: "100%",
      backgroundColor: theme.palette.primary.main,
    },
    form: {
        display: 'grid',
        margin: 'auto',
        width: '55%',
        minWidth: '300px',
        maxWidth: "550px",
    },
    formControl: {
        display: 'grid',
        gridTemplateColumns: "repeat(auto-fit, minmax(200px, 1fr))",
        gridGap: '15px'
    },
    button: {
        width: '50%',
        justifySelf: 'center',
        minWidth: '200px'
    },
    formGroup: {
        '& > *': {
            margin: '15px 0'
        }
    }
}));


const AccountSettings = () => {
    const classes = useStyles();
    const dispatch = useDispatch();
    const {id, isAuth, ...user} = useSelector(state => state.user.currentUser);

    const onSubmit = formData => {
        dispatch(updateUser(formData));
    };

    if(!isAuth) return <p>Loading...</p>;
    return (
        <div className={classes.wrapper}>
            <AccountSettingsForm onSubmit={onSubmit} user={user} classes={classes}/>
        </div>
    )
};


export default  withAuthRedirect(false)(AccountSettings);