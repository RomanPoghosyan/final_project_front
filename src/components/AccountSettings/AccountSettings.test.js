import React from "react";
import AccountSettings from "./AccountSettings";
import renderer from 'react-test-renderer';
import withAuthRedirect from "../../hoc/withAuthRedirect";

describe('AccountSettings component', () => {
    it('Test - should return null', () => {
        const NewComponent = withAuthRedirect(AccountSettings);
        const tree = renderer.create(<NewComponent/>).toJSON();
        expect(tree).toBe(null);
    });
});