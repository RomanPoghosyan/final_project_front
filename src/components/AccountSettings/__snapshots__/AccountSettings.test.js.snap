// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountSettings component Test - to Match Snapshot Before 1`] = `null`;
